[Fu Yiqiao] iP

fuyiqiao opened the pull request:
### Ballsorter

Ballsorter allows you to 🤗
1. set **todos**
2. set _deadlines_ 
3. set **events**!

> Have fun! - Me [check this out!](https://github.com/fuyiqiao)

This chatbot helps you to
- [ ] become happier 
- [ ] become better

Sample driver code: 
```
public static void main(String[] args) {
        new Ballsorter().run();
}
```
- or add your own filepath! 
   `new Ballsorter(data/Ballsorter.txt)`

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

fuyiqiao committed:
Level 0

fuyiqiao committed:
Level 1

fuyiqiao committed:
Level 2

fuyiqiao committed:
Level 3

fuyiqiao committed:
A-Classes

Commit:
Level-4

Commit:
A-TextUiTesting

Commit:
Level-5

Commit:
Level-6

Commit:
Level-7

Commit:
Merge branch-Level-7

Commit:
Level-8

Commit:
Merge branch-Level-8

Commit:
A-MoreOOP

Commit:
A-Packages

Commit:
Merge add-gradle-support

Commit:
A-Gradle

Commit:
A-JUnit

Commit:
A-Jar

Commit:
A-JavaDoc

Commit:
A-CodingStandard

Commit:
Level-9

Commit:
Merge branch-Level-9

Commit:
Merge branch-A-CodingStandard

Commit:
resolve conflicts

Commit:
resolve conflicts

Commit:
A-CheckStyle

Commit:
Merge branch-A-CheckStyle

Commit:
Level-10

Commit:
Merge branch-Level-10

jellywaiyan commented on src/main/java/ballsorting/Ballsorter.java:
```diff
@@ -0,0 +1,204 @@
+/**
+ * Main class of the chatbot.
+ */
+public class Ballsorter extends Application {
```
Shouldn't class name be in PascalCase? (unless Ballsorter is just one word :o)

jellywaiyan commented on src/main/java/ballsorting/Ballsorter.java:
```diff
@@ -0,0 +1,204 @@
+    }
+
+    /**
+     * Creates a new instance of the chatbot.
```
Should have an empty line between desc and param section

jellywaiyan commented on src/main/java/ballsorting/Ballsorter.java:
```diff
@@ -0,0 +1,204 @@
+        return textToAdd;
+    }
+
+    /**
+     * You should have your own function to generate a response to user input.
+     * Replace this stub with your completed method.
+     */
+    protected String getResponse(String input) {
+        return "Duke heard: " + input;
```
Unedited portion? 😶‍🌫️

tiongMax replied:
Seems like the level-10 is not fully completed but good job for following along.

jellywaiyan commented on src/main/java/ballsorting/Deadline.java:
```diff
@@ -0,0 +1,38 @@
+package ballsorting;
```
I think there should be a space between package and import statements? But am not sure whether it's included in our coding standards

tiongMax replied:
yep, there should be a space between package and import

jellywaiyan reviewed (commented):
Looks very good to me, just some very minor style fixes that I feel would make it cleaner. Good job :o

tiongMax commented on src/main/java/ballsorting/DialogBox.java:
```diff
@@ -0,0 +1,44 @@
+public class DialogBox extends HBox {
+    private Label text;
+    private ImageView displayPicture;
+    public DialogBox(Label l, ImageView iv) {
```
It will be better to separate fields and constructor by space.

tiongMax reviewed (commented):
Overall, this is an amazing work. It will be better if the minor mistakes are fixed. Keep it up!

Commit:
Added assertions at various points in the code.

Added assertions to ensure that Task objects are not null.

Added assertions that descriptions for Tasks are not empty Strings at
various points.
Added assertions for events that the End time does not occur before
the Start time.

Commit:
Code quality

Extracted the error messages into a separate class.

Replaced some magic numbers with expressions that explain what they mean.

fuyiqiao committed:
Merge pull request #1 from fuyiqiao/branch-A-Assertions

Added assertions at various points in the code.

Commit:
Merge branch 'master' into branch-A-CodeQuality

Update CodeQuality branch.

fuyiqiao committed:
Merge pull request #2 from fuyiqiao/branch-A-CodeQuality

Branch a code quality

Commit:
C-DetectDuplicates: added the functionality of detecting duplicated tasks

Commit:
Merge branch-C-DetectDuplicates

fuyiqiao committed:
Update README.md

fuyiqiao committed:
Update README.md

fuyiqiao committed:
Update README.md

Commit:
A-MoreErrorHandling

Commit:
Merge branch 'master' of github.com:fuyiqiao/ip

Commit:
A-MoreErrorHandling

Commit:
-

Commit:
Added javadocs

Commit:
fixed file path issues
